src/picklescan/scanner.py

```diff
@@ -307,12 +327,35 @@ def scan_pickle_bytes(data: IO[bytes], file_id, multiple_pickles=True) -> ScanRe
     return _build_scan_result_from_raw_globals(raw_globals, file_id)
 
 
+# XXX: it appears there is not way to get the byte stream for a given file within the 7z archive and thus forcing us to unzip to disk before scanning
```

Comment:
It looks like py7zr's way to get byte streams is by deriving custom classes from `Py7zIO` and `WriterFactory`: https://py7zr.readthedocs.io/en/latest/advanced.html
That feels a bit like an overkill here though, and extracting to disk sounds fine.

Reply:
Saw that too, but didn't figure out a way to get a per-file byte stream, will check again.

Reply:
I would propose to skip. The current trade-off looks good enough for what the code needs to do.